api/src/main/java/org/eclipse/microprofile/config/ConfigValue.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.config;
+
+/**
+ * The ConfigValue holds additional information after the lookup of a configuration.
+ * <p>
+ *
+ * Holds information about the configuration name, configuration value, the
+ * {@link org.eclipse.microprofile.config.spi.ConfigSource} name from where
+ * the configuration was loaded and the ordinal of the {@link org.eclipse.microprofile.config.spi.ConfigSource}.
+ * <p>
+ *
+ * This is used together with {@link Config} to expose the configuration lookup metadata.
+ *
+ * @author <a href="mailto:radcortez@yahoo.com">Roberto Cortez</a>
+ */
+public interface ConfigValue {
+    /**
+     * The name of the lookup property.
+     *
+     * @return the name of the lookup property.
+     */
+    String getName();
+
+    /**
+     * The value of the config lookup.
+     *
+     * @return the raw value of the config lookup or {@code null} if the property could not be found.
```

Comment:
This should clarify if this is the value at the point in time the `ConfigValue` object was looked up or if it is the value when `getValue()` is called.

Reply:
Yes, it needs to be when the `ConfigValue` was looked up, because of the other methods. Let me add a clarification for that.

Not so sure about adding a `getTime()` method.

Reply:
I don't think we should add `getTime()` method as this `getValue()` is no difference from the `config.getValue()`. The return result of the `getValue()` should always be the updated value not cached value if a runtime supports dynamic config. The ConfigValue should be treated as a proxy.

Reply:
Do you mean that when you call `getValue` on `ConfigValue` is when the lookup is performed? What about the other methods? What would happen if you call `getSourceName` before calling `getValue?

Reply:
As mentioned in our call earlier, I am fine with making configValue as immutable as getConfigVaue will retrieve the updated value.